2024/11/4/cancellable-interval/cancellable-interval.test.ts

```
@@ -0,0 +1,47 @@
+import { describe, expect, test } from "vitest";
+import setCancellableInterval from "./cancellable-interval";
+
+describe("setCancellableInterval", () => {
+  test("returns a function", () => {
+    expect(typeof setCancellableInterval(() => {})).toBe("function");
+  });
+
+  test("cancelled immediately", async () => {
+    expect.assertions(2);
+
+    const promise = new Promise((resolve: Function) => {
```

Biome

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)